src/features/defi/providers/cosmos/components/CosmosManager/Claim/ClaimConfirm.tsx

```diff
@@ -0,0 +1,211 @@
+import { useStakingAction } from 'plugins/cosmos/hooks/useStakingAction/useStakingAction'
+import { getStakingFees } from 'plugins/cosmos/utils'
+import { useEffect, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { Row } from 'components/Row/Row'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { selectAssetById, selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type ClaimConfirmProps = {
+  assetId: AssetId
+  amount?: string
+  contractAddress: string
+  chainId: ChainId
+  onBack: () => void
+}
+
+export const ClaimConfirm = ({
+  assetId,
+  amount,
+  contractAddress,
+  chainId,
+  onBack,
+}: ClaimConfirmProps) => {
+  const [userAddress, setUserAddress] = useState<string>('')
+  const [estimatedGas, setEstimatedGas] = useState<string>('0')
+  const [loading, setLoading] = useState<boolean>(false)
+  const chainAdapterManager = getChainAdapterManager()
+  const { state: walletState } = useWallet()
+  const translate = useTranslate()
+  const claimAmount = bnOrZero(amount).toString()
+  const history = useHistory()
+
+  // Asset Info
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Cosmos, // TODO: Programmatic
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  const toast = useToast()
+
+  const { handleStakingAction } = useStakingAction()
+
+  const handleConfirm = async () => {
+    if (!walletState.wallet || !contractAddress || !userAddress) return
+    setLoading(true)
+
+    const { gasLimit, gasPrice } = await getStakingFees(asset, feeMarketData.price)
+
+    try {
+      const broadcastTxId = await handleStakingAction({
+        asset,
+        validator: contractAddress,
+        chainSpecific: {
+          gas: gasLimit,
+          fee: bnOrZero(gasPrice).times(`1e+${asset?.precision}`).toString(),
+        },
+        value: bnOrZero(claimAmount).times(`1e+${asset.precision}`).toString(),
+        action: StakingAction.Claim,
+      })
+      history.push('/status', {
+        txid: broadcastTxId,
+        assetId,
+        amount,
+        userAddress,
+        estimatedGas,
+        chainId,
+      })
+    } catch (error) {
+      console.error('ClaimWithdraw:handleConfirm error', error)
```

Comment:
moduleLogger

Reply:
Also from the Foxy implementation, will add it in the same stacked PR

Reply:
https://github.com/shapeshift/web/pull/2323